packages/core/contracts/snapshot-helpers/SnapshotVotingPower.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.16;
+
+/**
+ * @title VotingInterface
+ * @notice Minimum interface required to interact with VotingV2 contract.
+ */
+interface VotingInterface {
+    struct VoterStake {
+        uint128 stake;
+        uint128 pendingUnstake;
+        uint128 rewardsPaidPerToken;
+        uint128 outstandingRewards;
+        int128 unappliedSlash;
+        uint64 nextIndexToProcess;
+        uint64 unstakeTime;
+        address delegate;
+    }
+
+    function voterStakes(address)
+        external
+        view
+        returns (
+            uint128,
+            uint128,
+            uint128,
+            uint128,
+            int128,
+            uint64,
+            uint64,
+            address
+        );
+
+    function getVoterFromDelegate(address) external view returns (address);
+}
+
+/**
+ * @title SnapshotVotingPower
+ * @notice Helper contract to support offchain voting with Snapshot.
+ */
+contract SnapshotVotingPower {
+    VotingInterface public immutable votingV2 = VotingInterface(0x004395edb43EFca9885CEdad51EC9fAf93Bd34ac);
+
+    /**
+     * @notice This is only used by Snapshot to calculate voting power and does not represent transferable tokens.
+     * @param user address of the user for whom to calculate voting power.
+     * @return uint256 value of user's voting power based on staked UMA at DVM2.0.
+     **/
+    function balanceOf(address user) external view returns (uint256) {
+        address voter = votingV2.getVoterFromDelegate(user);
+        (uint128 stake, , , , , , , address delegate) = votingV2.voterStakes(voter);
+
+        // Avoid double counting in case of stake delegation.
+        if (delegate != address(0) && user != delegate) return 0;
+
```

Comment:
Does the snapshot system handle the case where a user has staked tokens _and_ a balance in their address? Or should we add that here?

Reply:
Snapshot can account for unstaked balance in a separate strategy, but we are deliberately excluding it so that snapshot voting power is as close as possible to DVM voting power (except for updated slashing trackers that would require forking simulation that is not supported in Snapshot)